backend/player.js

```diff
@@ -0,0 +1,147 @@
+            propertyJSONs.push(p.toJSON());
+        });
+
+        return {
+            "name": this.name,
+            "money": this.money,
+            "properties": propertyJSONs,
+            "busTickets": this.busTickets,
+            "forward": this.forward,
+            "location": this.location,
+            "specialCards": this.specialCards
+        }
+    }
+
+    /* Calculates the net worth of the player
+     * 
+     * @return int net worth of the player
+     */
+    getNetWorth() {
+        let worth = this.money
+
+        this.properties.forEach(property => {
+            worth += property.getValue()
+        });
+
+        return worth
+    }
+
+    /* Determines if the player can afford to buy item of amt, but doesn't change amount of money
+     * @param amt - numerical amount of money to check affordability with
+     * @return true if player can afford, false otherwise
+     */
+    canAfford(amt){
+        return this.money - amt >= 0
+    }
+
+    /* The player acquires the specified property.
+     * @param property Object of the property
+     */
+    gainProperty(property) {
+        this.properties.push(property)
+    }
+
+    /* The player loses the specified property.
+     * @param property Object of the property
+     */
+    loseProperty(property) {
+        let index = this.properties.indexOf(property)
+        this.properties.splice(index, 1)
+    }
+
+    /* The player's money changes by a certain amount.
+     * @param amt number amount of money to change by (can be pos/neg)
+     *
+     * @return true if player still has money left, false if negative (bankrupt)
+     */
+    deltaMoney(amt) {
+        this.money += amt
+        return this.money >= 0
+    }
+
+    /* The player acquires the specified bus pass.
+     * @param pass String name of the bus pass
+     */
+    gainBusPass(pass) {
+        if (pass.includes('expire')){
+            this.busTickets = {}
+            this.busTickets[pass] = 1
+        }
+        else if (this.busTickets.hasOwnProperty(pass))
+            this.busTickets[pass] += 1
+        else
+            this.busTickets[pass] = 1
+
+        
+    }
+
+    /* The player uses the specified bus pass and loses it.
+     * @param pass String name of the bus pass
+     */
+    useBusPass(pass) {
+        // actual ticket usage handled by some other class
+        this.busTickets[pass] -= 1
+        if (this.busTickets[pass] === 0)
+            delete this.busTickets[pass]
+
+    }
+
+    /* The player acquires the specified chance/community chest card.
+     * @param card Object name of the card
+     */
+    gainSpecialCard(card) {
+        if (this.specialCards.hasOwnProperty(card))
+            this.specialCards[card] += 1
+        else
+            this.specialCards[card] = 1
+    }
+
+    /* The player uses the specified chance/community chest card and loses it.
+     * @param card Object name of the card
+     */
+    useSpecialCard(card) {
```

Comment:
I think this function and other functions should probably return true if the requested action was successful and false if it was not (ie. card not found). We should do this probably for all of the other functions also. Then we need to update the tests with just assert(<operation>) or assert(!<failed operation>).